dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/ClientBase.cs

```diff
@@ -337,22 +317,28 @@ private static ChatCompletionsOptions CreateChatCompletionsOptions(ChatRequestSe
             }
         }
 
-        foreach (ChatHistory.Message message in chat.Messages)
+        foreach (var message in chatHistory)
         {
-            var role = message.AuthorRole switch
-            {
-                ChatHistory.AuthorRoles.User => ChatRole.User,
-                ChatHistory.AuthorRoles.Assistant => ChatRole.Assistant,
-                ChatHistory.AuthorRoles.System => ChatRole.System,
-                _ => throw new ArgumentException($"Invalid chat message author: {message.AuthorRole:G}")
-            };
+            ValidateChatAuthors(message.Role, out var validRole);
 
-            options.Messages.Add(new ChatMessage(role, message.Content));
+            options.Messages.Add(new ChatMessage(validRole, message.Content));
         }
 
         return options;
     }
 
+    private static void ValidateChatAuthors(AuthorRole role, out ChatRole validRole)
+    {
+        validRole = new ChatRole(role.Label);
+
+        if (validRole != ChatRole.User &&
+            validRole != ChatRole.System &&
+            validRole != ChatRole.Assistant)
+        {
+            throw new ArgumentException($"Invalid chat message author: {role}");
```

Comment:
Small improvement in exception message:
```suggestion
            throw new ArgumentException($"Invalid chat message author role: {role}");
```